Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h

```diff
@@ -1635,6 +1636,300 @@ class Variational_shape_approximation {
+          g_vertex_descriptor g_v2 = get(tm_to_g_vertices, v2);
+          CGAL_assertion(graph[g_v2]==v2);
+
+// HACKY (fixes case of the issue but breaks the testsuite)
+if ( (!get(is_patch_border, h)) &&
+      get(closest_anchor_ids, v2) != CGAL_VSA_INVALID_TAG &&
+      get(closest_anchor_ids, v2)!=get(closest_anchor_ids, v)) continue;
+
+
+          g_edge_descriptor e = add_edge(nv, g_v2, graph).first;
+          graph[e] =  CGAL::approximate_sqrt(
+                        CGAL::squared_distance(m_vpoint_map[source(h, *m_ptm)],
+                                               m_vpoint_map[v]));
+
+#ifdef CGAL_SURFACE_MESH_APPROXIMATION_CDT_DEBUG
+          debug_out2 << "2 " << m_vpoint_map[source(h, *m_ptm)] <<
+                       " " << m_vpoint_map[v] << "\n";
+#endif
+
+        }
+      }
+
+      for (vertex_descriptor v : vertices_used)
+      {
+        g_vertex_descriptor g_v1 = get(tm_to_g_vertices, v);
+        for (halfedge_descriptor h : halfedges_around_target(v, *m_ptm))
+        {
+          vertex_descriptor v2 = source(h, *m_ptm);
+          if (vertices_used.count(v2)==0 || v2 < v) continue;
+          g_vertex_descriptor g_v2 = get(tm_to_g_vertices, v2);
+// HACKY (fixes case of the issue but breaks the testsuite)
+if ( (!get(is_patch_border, h)) &&
+     get(closest_anchor_ids, v2) != CGAL_VSA_INVALID_TAG &&
+     get(closest_anchor_ids, v) != CGAL_VSA_INVALID_TAG &&
+     get(closest_anchor_ids, v2)!=get(closest_anchor_ids, v) ) continue;
+
+          g_edge_descriptor e = add_edge(g_v1, g_v2, graph).first;
+          graph[e] =  CGAL::approximate_sqrt(
+                        CGAL::squared_distance(m_vpoint_map[v],
+                                               m_vpoint_map[v2]));
+#ifdef CGAL_SURFACE_MESH_APPROXIMATION_CDT_DEBUG
+          debug_out2 << "2 " << m_vpoint_map[v] <<
+                       " " << m_vpoint_map[v2] << "\n";
+#endif
+
+        }
+      }
+
+#ifdef CGAL_SURFACE_MESH_APPROXIMATION_CDT_DEBUG
+      debug_out2.close();
+#endif
+
+      std::vector<g_vertex_descriptor> pred(num_vertices(graph), boost::graph_traits<Graph>::null_vertex());
+      boost::dijkstra_shortest_paths(graph, super_vertex,
+          boost::predecessor_map(CGAL::make_property_map(pred)).weight_map(get(boost::edge_bundle, graph)));
+
+      // extract info in predecessor_map: TODO avoid to be too recursive
+      for(g_vertex_descriptor gv : make_range(vertices(graph)))
+      {
+        if (gv==super_vertex) continue;
+        CGAL_assertion( graph[gv] != boost::graph_traits<TriangleMesh>::null_vertex() );
+        if ( get(closest_anchor_ids, graph[gv]) != CGAL_VSA_INVALID_TAG ) continue;
+
+        g_vertex_descriptor igv=gv;
+        while (pred[igv]!=super_vertex)
+        {
+#ifdef CGAL_SURFACE_MESH_APPROXIMATION_CDT_DEBUG
+          if( igv == pred[igv] )
+          {
+            CGAL_assertion(igv==gv);
+            std::cerr << "ISSUE with " << m_vpoint_map[graph[gv]] << "\n";
```

Comment:
Should that line remain?

Reply:
From the description: "This needs to be discussed with @palliez and cleaned up"